src/App.tsx

```diff
@@ -1,15 +1,57 @@
-import React from 'react';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { DropDownMenu } from './components/DropdownMenu';
+import { PersonData } from './types/PersonData';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectPerson, setSelectPerson] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selectedPersonData, setSelectedPersonData] =
+    useState<PersonData | null>(null);
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 1000), []);
```

Comment:
According to our checklist, we need to make sure that the filter won't be called if the user entered spaces only. Currently, if the user inputs spaces, the filter will be applied and this is not the desired behavior. You should trim the input value before passing it to the `applyQuery` function.